app/models/automation_rule.rb

```diff
@@ -20,9 +20,13 @@
 class AutomationRule < ApplicationRecord
   include Rails.application.routes.url_helpers
 
+  MAX_ERROR_THRESHOLD = 3
```

Comment:
can't we use reauthorizable concern for this case as well ?  seems like lot of logic is pretty common

Reply:
@sojan-official I've made the changes, let me test it out properly. I'll let you know once it's ready for review